src/cuda_eval.cpp

```diff
@@ -802,6 +842,8 @@ static void jitc_cuda_render_scatter(const Variable *v,
     const char *op = reduce_op_name[v->literal];
     const ThreadState *ts = thread_state_cuda;
 
+    if (jitc_is_half(value) && v->literal != 0)
```

Comment:
This is not a nice way to catch this special case because it will crash the Python session. Can we already do this in `op.cpp` where the associated scatter variable would be created?